src/ralph/admin/tests/tests_fields.py

```diff
@@ -0,0 +1,90 @@
+# -*- coding: utf-8 -*-
+from django import forms
+from django.core.exceptions import ValidationError
+from django.test import SimpleTestCase
+
+from ralph.admin.fields import MultilineField, MultivalueFormMixin
+
+
+class SimpleTestForm(MultivalueFormMixin, forms.Form):
+    multivalue_fields = ['sn', 'barcode']
+    sn = MultilineField()
+    barcode = MultilineField()
+
+
+class OneRequiredTestForm(MultivalueFormMixin, forms.Form):
+    one_of_mulitvalue_required = ['sn', 'barcode']
+    multivalue_fields = ['sn', 'barcode']
+    sn = MultilineField()
+    barcode = MultilineField()
+
+
+class MultiValueFormTest(SimpleTestCase):
+    def test_works_for_single_value_each(self):
+        data = {
+            'sn': 'sn1',
+            'barcode': 'bc1',
+        }
+        form = SimpleTestForm(data)
+        self.assertTrue(form.is_valid())
+
+    def test_works_for_multi_value_each(self):
+        data = {
+            'sn': 'sn1, sn2, sn3',
+            'barcode': 'bc1, bc2, bc3',
+        }
+        form = SimpleTestForm(data)
+        self.assertTrue(form.is_valid())
+
+    def test_not_valid_when_different_count(self):
+        data = {
+            'sn': 'sn1',
+            'barcode': 'bc1, bc2',
+        }
+        form = SimpleTestForm(data)
+        self.assertFalse(form.is_valid())
+        self.assertIn('sn', form.errors)
+        self.assertIn('barcode', form.errors)
+
+    def test_not_valid_when_none_multivalue_passed(self):
+        data = {
+            'sn': 'sn1,,sn3',
+            'barcode': 'br1,,br3',
+        }
+        form = OneRequiredTestForm(data)
+        self.assertFalse(form.is_valid())
+        self.assertIn('sn', form.errors)
+        self.assertIn('barcode', form.errors)
+
+
+class MultilineFieldTest(SimpleTestCase):
```

Comment:
could you write additional test when none of required fields is filled (fields have blank values)?